src/keyring/api.rs

```diff
@@ -0,0 +1,500 @@
+/*!
+GNOME Keyring format low level api
+
+Only use this if you know what you are doing.
+
+### To Do
+
+- Order user calls
+- Keep proxis around
+- Make more things async
+*/
+
+use async_std::prelude::*;
+
+use async_std::{fs, io, path::Path};
+use cipher::{
+    block_padding::Pkcs7, crypto_common::rand_core, BlockDecryptMut, BlockEncryptMut,
+    BlockSizeUser, IvSizeUser, KeyIvInit,
+};
+use digest::OutputSizeUser;
+use hmac::Mac;
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::path::PathBuf;
+use zbus::zvariant::{self, Type};
+use zeroize::{Zeroize, ZeroizeOnDrop, Zeroizing};
+
+const SALT_SIZE: usize = 32;
+const ITERATION_COUNT: u32 = 100000;
+
+const FILE_HEADER: &[u8] = b"GnomeKeyring\n\r\0\n";
+const FILE_HEADER_LEN: usize = FILE_HEADER.len();
+
+const MAJOR_VERSION: u8 = 1;
+const MINOR_VERSION: u8 = 0;
+
+type MacAlg = hmac::Hmac<sha2::Sha256>;
+type EncAlg = cbc::Encryptor<aes::Aes128>;
+type DecAlg = cbc::Decryptor<aes::Aes128>;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug)]
```

Comment:
I would move the Error type into a separate file. along with the keyring stuff to the portal module